config/src/main/java/com/typesafe/config/impl/ConfigNodeComplexValue.java

```diff
@@ -0,0 +1,89 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config.impl;
+
+import com.typesafe.config.ConfigNode;
+
+import java.util.*;
+
+final class ConfigNodeComplexValue extends AbstractConfigNodeValue {
+    final private ArrayList<AbstractConfigNode> children;
+
+    ConfigNodeComplexValue(Collection<AbstractConfigNode> children) {
+        this.children = new ArrayList(children);
+    }
+
+    public Iterable<AbstractConfigNode> children() {
+        return children;
+    }
+
+    protected Collection<Token> tokens() {
+        ArrayList<Token> tokens = new ArrayList();
+        for (AbstractConfigNode child : children) {
+            tokens.addAll(child.tokens());
+        }
+        return tokens;
+    }
+
+    protected ConfigNodeComplexValue changeValueOnPath(Path desiredPath, AbstractConfigNodeValue value) {
+        ArrayList<AbstractConfigNode> childrenCopy = new ArrayList(children);
+        boolean replaced = value == null;
+        ConfigNodeKeyValue node;
+        Path key;
+        for (int i = children.size() - 1; i >= 0; i--) {
+            if (!(children.get(i) instanceof ConfigNodeKeyValue)) {
+                continue;
+            }
+            node = (ConfigNodeKeyValue)children.get(i);
+            key = node.key().value();
+            if (key.equals(desiredPath)) {
+                if (!replaced) {
```

Comment:
it's a nice practice to avoid combining negation and if/else (do `if (replaced) { remove(i) } else {}`)